src/utils/AssayGeneratorMetabo.cpp

```diff
@@ -0,0 +1,523 @@
+      if (s_it2 == spectra.end())
+      {
+        LOG_WARN << "No MS1 Spectrum was found to the specific precursor" << std::endl;
+        continue;
+      }
+
+      const MSSpectrum& precursor_spectrum = *s_it2;
+      Int mono_index = getHighestIntensityPeakInMZRange(test_mz, precursor_spectrum, left_tolerance, right_tolerance);
+      if (mono_index == -1)
+      {
+        LOG_WARN << "No precusor peak in MS1 spectrum found." << std::endl;
+        continue;
+      }
+      const Peak1D& max_mono_peak = precursor_spectrum[mono_index];
+      precursor[0].setMZ(max_mono_peak.getMZ());
+      precursor[0].setIntensity(max_mono_peak.getIntensity());
+    }
+  }
+
+  ExitCodes main_(int, const char **) override
+  {
+    //-------------------------------------------------------------
+    // Parsing parameters
+    //-------------------------------------------------------------
+
+    String in = getStringOption_("in");
+    String id = getStringOption_("in_id");
+    String out = getStringOption_("out");
+    String method = getStringOption_("method");
+    double precursor_mz_tol = getDoubleOption_("precursor_mz_tolerance");
+    double precursor_rt_tol = getDoubleOption_("precursor_rt_tolerance");
+    double cosine_sim_threshold = getDoubleOption_("cosine_similarity_threshold");
+    bool method_consensus_spectrum = method == "consensus_spectrum" ? true : false;
+
+    unsigned int hull_size_filter = getIntOption_("filter_by_convex_hulls");
+    double transition_threshold = getDoubleOption_("transition_threshold");
+    bool use_known_unknowns = getFlag_("use_known_unknowns");
+
+    // load mzML
+    MzMLFile mzml;
+    PeakMap spectra;
+    mzml.load(in, spectra);
+
+    // load featurexml
+    FeatureXMLFile fxml;
+    FeatureMap feature_map;
+    fxml.load(id, feature_map);
+
+    // check if correct featureXML is given and set use_known_unkowns parameter if no id information is available
+    const std::vector<DataProcessing>& processing = feature_map.getDataProcessing();
+    for (auto it = processing.begin(); it != processing.end(); ++it)
+    {
+      if (it->getSoftware().getName() == "FeatureFinderMetabo")
+      {
+        // check if convex hulls parameter was used in the FeatureFinderMetabo
+        if (it->getMetaValue("parameter: algorithm:ffm:report_convex_hulls") != "true")
+        {
+          throw Exception::InvalidParameter(__FILE__,
+                                            __LINE__,
+                                            OPENMS_PRETTY_FUNCTION,
+                                            "Please provide a valid feature XML file with reported convex hulls.");
+        }
+        // if id information is missing set use_known_unknowns to true
+        if (feature_map.getProteinIdentifications().empty())
+        {
+          use_known_unknowns = true;
+        }
+      }
+    }
+
+    // annotate precursor mz and intensity
+    annotatePrecursorIntensity(spectra, 0.2, 0.2);
```

Comment:
these values should be adjustable, for TOF and Orbitrap data you would want to have different tolerances here!

Reply:
should have precursor_mz_tolarance as a variable (will change that)